lib/utils/matrix_sdk_extensions/flutter_matrix_dart_sdk_database/builder.dart

```diff
@@ -52,63 +51,61 @@ Future<MatrixSdkDatabase> _constructDatabase(Client client) async {
+
+    fileStoragePath = await getApplicationSupportDirectory();
+
+    database = await dbFactory.openDatabase(
+      '${fileStoragePath.path}/${client.clientName}',
+      options: OpenDatabaseOptions(
+        version: 1,
+        // pass
+        onConfigure:
+            password == null ? null : (db) => _setPragmaKey(db, password),
       ),
-      maxFileSize: 1024 * 1024 * 10,
-      fileStoragePath: path,
-      deleteFilesAfterDuration: const Duration(days: 30),
     );
-  }
+  } else {
+    final path = await getApplicationSupportDirectory();
+    final sqlFilePath = '$path/${client.clientName}.sqlite';
 
-  final path = await getDatabasesPath();
-  const passwordStorageKey = 'database_password';
-  String? password;
+    // migrating from petty unreliable `getDatabasePath`
+    // See : https://pub.dev/packages/sqflite_common_ffi#limitations
+    await _migrateLegacyLocation(sqlFilePath, client.clientName);
 
-  try {
-    // Workaround for secure storage is calling Platform.operatingSystem on web
-    if (kIsWeb) throw MissingPluginException();
-
-    const secureStorage = FlutterSecureStorage();
-    final containsEncryptionKey =
-        await secureStorage.read(key: passwordStorageKey) != null;
-    if (!containsEncryptionKey) {
-      final rng = Random.secure();
-      final list = Uint8List(32);
-      list.setAll(0, Iterable.generate(list.length, (i) => rng.nextInt(256)));
-      final newPassword = base64UrlEncode(list);
-      await secureStorage.write(
-        key: passwordStorageKey,
-        value: newPassword,
-      );
-    }
-    // workaround for if we just wrote to the key and it still doesn't exist
-    password = await secureStorage.read(key: passwordStorageKey);
-    if (password == null) throw MissingPluginException();
-  } on MissingPluginException catch (_) {
-    const FlutterSecureStorage()
-        .delete(key: passwordStorageKey)
-        .catchError((_) {});
-    Logs().i('Database encryption is not supported on this platform');
-  } catch (e, s) {
-    const FlutterSecureStorage()
-        .delete(key: passwordStorageKey)
-        .catchError((_) {});
-    Logs().w('Unable to init database encryption', e, s);
+    fileStoragePath = await getTemporaryDirectory();
+    database = await openDatabase(
+      sqlFilePath,
+      password: password,
+    );
   }
 
   return MatrixSdkDatabase(
     client.clientName,
-    database: await openDatabase(
-      '$path/${client.clientName}',
-      password: password,
-    ),
+    database: database,
     maxFileSize: 1024 * 1024 * 10,
-    fileStoragePath: await getTemporaryDirectory(),
+    fileStoragePath: fileStoragePath,
     deleteFilesAfterDuration: const Duration(days: 30),
   );
 }
+
+Future<void> _migrateLegacyLocation(
+  String sqlFilePath,
+  String clientName,
+) async {
+  final oldPath = getDatabasesPath();
+  final maybeOldFIle = File('$oldPath/$clientName.sqlite');
+  if (await maybeOldFIle.exists()) {
```

Comment:
Should check before if the two path strings are already equal. Otherwise it would copy the database every time the app opens